File: 2015/01/zig/src/main.zig
const std = @import("std");
const BUF_SIZE = 1024;

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();
    const stdout = std.io.getStdOut().writer();

    var floor: i64 = 0;
    var current_index: usize = 0;
    var first_basement_index: usize = 0;
    var buf: [BUF_SIZE]u8 = undefined;
    while (true) {
        const read_bytes = try stdin.read(&buf);
        if (read_bytes == 0) {
            break;
        }
        for (buf[0..read_bytes]) |c| {
            if (c == '(') floor += 1;
            if (c == ')') floor -= 1;
            current_index += 1;
            if (floor == -1 and first_basement_index == 0) {
                first_basement_index = current_index;
            }
        }
    }
    try stdout.print("{}\n{}\n", .{ floor, first_basement_index });
}

File: 2015/01/c/main.c
#include <unistd.h>

#define BUF_SIZE 1024

void putchar(char c) {
  write(1, &c, 1);
}

void putnbr(int n) {
  if (n < 0) {
    putchar('-');
    n = -n;
  }
  if (n < 10) {
    putchar(n + '0');
  } else {
    putnbr(n / 10);
    putchar(n % 10 + '0');
  }
}

void putnbrln(int n) {
  putnbr(n);
  putchar('\n');
}

int main() {
  int floor = 0;
  int current_index = 0;
  int first_basement_index = 0;
  char buffer[BUF_SIZE];
  ssize_t bytes_read;

  while ((bytes_read = read(STDIN_FILENO, buffer, BUF_SIZE))) {
    for (int i = 0; i < bytes_read; i++) {
      char c = buffer[i];
      if (c == '(') floor++;
      if (c == ')') floor--;
      current_index++;
      if (floor == -1 && first_basement_index == 0) {
        first_basement_index = current_index;
      }
    }
  }

  putnbrln(floor);
  putnbrln(first_basement_index);
}
